src/pydisagg/ihme/splitter/cat_splitter.py

```diff
@@ -0,0 +1,632 @@
+        """
+        List of value fields (attributes).
+
+        Returns
+        -------
+        List[str]
+            List containing 'val'.
+        """
+        return ["val"]
+
+    def apply_prefix(self) -> dict:
+        """
+        Create a mapping to rename columns with the specified prefix.
+
+        Returns
+        -------
+        dict
+            Mapping from original column names to prefixed column names.
+        """
+        return {
+            self.val: f"{self.prefix}{self.val}",
+            self.target: self.target,  # Do not prefix the target column
+            **{idx: idx for idx in self.index},  # Keep index columns unchanged
+        }
+
+
+class CatSplitter(BaseModel):
+    """
+    Class for splitting categorical data based on pattern and population data.
+
+    Parameters
+    ----------
+    data : CatDataConfig
+        Configuration for the data DataFrame.
+    pattern : CatPatternConfig
+        Configuration for the pattern DataFrame.
+    population : CatPopulationConfig
+        Configuration for the population DataFrame.
+    """
+
+    data: CatDataConfig
+    pattern: CatPatternConfig
+    population: CatPopulationConfig
+
+    def model_post_init(self, __context: Any) -> None:
+        """
+        Perform extra validation after model initialization.
+
+        Raises
+        ------
+        ValueError
+            If the match criteria in the pattern or population do not match the data.
+        """
+        if not set(self.pattern.index).issubset(self.data.index):
+            raise ValueError(
+                "Match criteria in the pattern must be a subset of the data"
+            )
+        if not set(self.population.index).issubset(
+            self.data.index + self.pattern.index
+        ):
+            raise ValueError(
+                "Match criteria in the population must be a subset of the data and the pattern"
+            )
+        # Check that the 'target' column in pattern and population matches data
+        if self.pattern.target != self.data.target:
+            raise ValueError(
+                "The 'target' column in pattern must match the 'target' column in data"
+            )
+        if self.population.target != self.data.target:
+            raise ValueError(
+                "The 'target' column in population must match the 'target' column in data"
+            )
+
+    def create_ref_return_df(self, data: DataFrame) -> tuple[DataFrame, DataFrame]:
+        """
+        Create reference and return DataFrames.
+
+        Parameters
+        ----------
+        data : DataFrame
+            The input data DataFrame.
+
+        Returns
+        -------
+        tuple[DataFrame, DataFrame]
+            A tuple containing:
+            - ref_df: DataFrame with original data, exploded if necessary.
+            - data: DataFrame with required columns and identifiers.
+        """
+        ref_df = data.copy()
+        ref_df["orig_pyd_id"] = range(
```

Comment:
I really want adding the `row_id` to be in a separate PR. Here let's still work with the assumption that `data.index` uniquely define a row.